expo-maps/src/Maps.tsx

```diff
@@ -17,13 +17,84 @@ const defaultNativeExpoMapViewProps: DefaultNativeExpoMapViewProps = {
 };
 
 export class ExpoMap extends React.Component<ExpoMapViewProps> {
+  private mapChildren() {
+    const childrenArray = React.Children.map(this.props.children, (child) => {
+      if (!isSimpleType(child)) {
+        if (instanceOfMarker(child)) {
+          return {
+            type: 'marker',
+            latitude: child.props.latitude,
+            longitude: child.props.longitude,
+          } as MarkerObject;
+        }
+      }
+      warnIfChildIsIncompatible(child);
+      return null;
+    });
+
+    return {
+      markers: childrenArray
+        ? childrenArray.filter((e) => e.type === 'marker')
+        : [],
+    };
+  }
+
   render() {
+    const childrenObj = this.mapChildren();
+
     let NativeExpoMapView = NativeExpoGoogleMapsView;
     if (Platform.OS == 'ios' && this.props.provider == 'apple') {
       NativeExpoMapView = NativeExpoAppleMapsView;
     }
+
     return (
-      <NativeExpoMapView {...defaultNativeExpoMapViewProps} {...this.props} />
+      <NativeExpoMapView
+        {...defaultNativeExpoMapViewProps}
+        {...this.props}
+        markers={childrenObj.markers}
+      />
     );
   }
 }
+
+export class Marker extends React.Component<MarkerProps> {
+  render() {
+    return null;
+  }
+}
+
+function instanceOfMarker(child: any): child is Marker {
+  if ('type' in child && String(child.type).includes('Marker')) {
+    return true;
+  }
+  return false;
```

Comment:
Maybe to have a greater protection we could check if a child contains every field it should contain, not only type